cypress/utils/plugins/security-analytics-dashboards-plugin/helpers.js

```diff
@@ -0,0 +1,298 @@
+        let numberOfBadges = $badge.length;
+        Cypress.log({
+          message: `Number of combo badges to clear: ${numberOfBadges}`,
+        });
+
+        cy.wrap(subject)
+          .parents('.euiComboBox__inputWrap')
+          .find('input')
+          .focus()
+          .pressBackspaceKey(numberOfBadges);
+      });
+  }
+);
+
+Cypress.Commands.add('validateDetailsItem', (label, value) => {
+  Cypress.log({
+    message: `Validate details item by label: ${label} and value: ${value}`,
+  });
+  return cy
+    .getElementByText('.euiFlexItem label', label)
+    .parent()
+    .siblings()
+    .contains(value);
+});
+
+Cypress.Commands.add('urlShouldContain', (path) => {
+  Cypress.log({ message: `Url should contain path: ${path}` });
+  return cy.url().should('contain', `#/${path}`);
+});
+
+Cypress.Commands.add(
+  'pressEnterKey',
+  {
+    prevSubject: true,
+  },
+  (subject) => {
+    Cypress.log({
+      message: 'Enter key pressed',
+    });
+    Cypress.automation('remote:debugger:protocol', {
+      command: 'Input.dispatchKeyEvent',
+      params: {
+        type: 'char',
+        unmodifiedText: '\r',
+        text: '\r',
+      },
+    });
+
+    return subject;
+  }
+);
+
+Cypress.Commands.add(
+  'pressBackspaceKey',
+  {
+    prevSubject: true,
+  },
+  (subject, numberOfPresses = 1) => {
+    Cypress.log({
+      message: 'Backspace key pressed',
+    });
+    _.times(numberOfPresses, () => {
+      Cypress.automation('remote:debugger:protocol', {
+        command: 'Input.dispatchKeyEvent',
+        params: {
+          type: 'rawKeyDown',
+          keyCode: 8,
+          code: 'Backspace',
+          key: 'Backspace',
+          windowsVirtualKeyCode: 8,
+        },
+      });
+      cy.wait(10);
+      Cypress.automation('remote:debugger:protocol', {
+        command: 'Input.dispatchKeyEvent',
+        params: {
+          type: 'rawKeyUp',
+          keyCode: 8,
+          code: 'Backspace',
+          key: 'Backspace',
+          windowsVirtualKeyCode: 8,
+        },
+      });
+    });
+  }
+);
+
+Cypress.Commands.add(
+  'validateTable',
+  {
+    prevSubject: true,
+  },
+  (subject, data) => {
+    Cypress.log({
+      message: 'Validate table elements',
+    });
+    return cy
+      .wrap(subject)
+      .should('be.visible')
+      .find('tbody')
+      .find('tr')
+      .then(($tr) => {
+        const length = data.length;
+        length && cy.get($tr).should('have.length', length);
+
+        cy.get($tr).within(($tr) => {
+          data.map((rowData) => {
+            rowData.forEach((tdData) => {
+              if (typeof tdData === 'string') {
+                tdData && cy.get($tr).find('td').contains(`${tdData}`);
+              } else {
+                // if rule is an object then use path
+                tdData && cy.get($tr).find('td').contains(`${tdData.path}`);
+              }
+            });
+          });
+        });
+      });
+  }
+);
+
+export const createDetector = (
```

Comment:
there is a customized command with the same name `createDetector` in https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/utils/plugins/security-analytics-dashboards-plugin/detectors.js#L9, not sure if it's confusing to use the same name but have different function

Reply:
Also this file has both customized commands and a util function, and this file is imported in commands.js, should we just have commands in this file?

Reply:
The util function is removed. I'll refactor it in some next PR.